server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java

```diff
@@ -9,9 +9,105 @@
 package org.elasticsearch.script.field;
 
 import java.math.BigInteger;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class BigIntegerField extends Field<java.math.BigInteger> {
 
+    /* ---- Conversion Class From Other Fields ----*/
+
+    /**
+     * Convert to a {@link BigIntegerField} from Long, Double or String Fields.
+     * Longs and Doubles are wrapped as BigIntegers.
+     * Strings are parsed as either Longs or Doubles and wrapped in a BigInteger.
+     */
+    public static final Converter<BigInteger, BigIntegerField> BigInteger;
+
+    static {
+        BigInteger = new Converter<BigInteger, BigIntegerField>() {
+            @Override
+            public BigIntegerField convert(Field<?> sourceField) {
+                if (sourceField instanceof LongField) {
+                    return LongField.convertLongToBigIntegerField((LongField) sourceField);
+                }
+
+                if (sourceField instanceof DoubleField) {
+                    return DoubleField.convertDoubleToBigIntegerField((DoubleField) sourceField);
```

Comment:
How about `DoubleField.toBigIntegerField`?

Reply:
I like that better too. I will make the change.

Reply:
Pushed the change for this.

Reply:
Changed the type conversion methods as well to this format.